docs/articles/instances.md

```diff
@@ -0,0 +1,26 @@
+# Instances
+
+## What is an Instance?
+Instances are the Marathon representation of [⇒Tasks](tasks.md) in Mesos. An Instance can exist even if there is currently no Task associated. Depending on the goal of a given Instance, Marathon will take actions to reach that goal: it might match offers in order to launch a Task or [⇒TaskGroup](taskGroups.md), or it might send kill requests to Mesos and eventually expunge the Instance from its state. See [⇒Instance Life Cycle](instanceLifeCycle.md) for more information.
+
+## How can I use Instances?
+Instances are created based on [⇒Service](services.md) specifications which can be either an [⇒App](apps.md) or a [⇒Pod](pods.md). Instances are implicitly created according to their App or Pod definition, which contains an instances property that designates the required amount of instances.
+
+## Which related concepts should I understand?
+When upgrading a Service during a [⇒Deployment](deployments.md) from one version to another, Marathon can create more instances than specified in either Service definition. It will do so if allowed via the [⇒upgradeStrategy](upgradeStrategy.md) configuration.
+When an instance is running on an agent disconnected from the cluster, it will eventually be reported as unreachable. According to the Service’s [⇒unreachableStrategy](unreachableStrategy.md), Marathon can schedule a replacement, even though the unreachable instance might still be running. When an unreachable instance becomes becomes reachable again, Marathon will scale down accordingly to the required amount of instances. It will decide over which instance shall be killed based on the [⇒killSelection](killSelection.md).
+
+## How should I use Instances?
+_(This section could point out how to use Service definitions so that the user doesn’t need one app per instance, thus circumventing known limitations.)_
+
+## Are there any limitations or things to consider?
+_(This section could point out known limitations wrt number of instances.)_
+
+## Links
+* [⇒Scaling Applications](apps-scaling.md)
+* [⇒Scaling Pods](pods-scaling.md)
+* [⇒Killing Instances](instance-operations.md#delete-instances)
+
+## Examples
+* [⇒Simple Application](examples/app-simple.json) definition configuration 3 instances
```

Comment:
Two questions:
- should we keep examples at the end of the article? Naively I'd expect an example in the text, where I introduce the topic
- what are good instances examples? Is it about `instances: 3` filed in the definition? Or is it about some `/v2/pods/{id}::status` or `/apps?embed=tasks` output?

Reply:
> should we keep examples at the end of the article?

I'm all for _linking_ examples in the text above. If we find any sort of collapsible boxes, or a second column etc, let's have examples in the text. The one thing I dislike is arbitrarily long code snippets in the docs, because they make articles very long and hard to parse IMO.

I'd like to distinguish between tutorial/recipe and documentation here: in a tutorial or recipe ("how to to xyz"), let's have all the steps and configs needed. In an article explaining what an instance is, I'd like to keep it short and easy to skim through.

A reason to link examples is also to have them as files we can use for tests.

> what are good instances examples?

I guess a spec that is as simple as possible, and only shows how instances are created through (currently) the instances field of a spec.